lib/linked_list.rb

```diff
@@ -7,39 +7,76 @@ def initialize
     @head = nil
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(1)
+  # Space complexity - O(1)
   def add_first(data)
-
+    @head = Node.new(data, @head)
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(1)
+  # Space complexity - O(1)
   def get_first
-
+    if @head.nil?
+      return @head
+    else
+      return @head.data
+    end
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(n) Where n is the number of nodes in the linked list
+  # Space complexity - O(n)
   def length
```

Comment:
Why do you think the space complexity is O(n)?  You are not doing recursion or creating another list.